libs/wingsdk/src/target-tf-aws/api.ts

```diff
@@ -0,0 +1,157 @@
+import { ApiGatewayResource } from "@cdktf/provider-aws/lib/api-gateway-resource";
+import { ApiGatewayRestApi } from "@cdktf/provider-aws/lib/api-gateway-rest-api";
+import { ApiGatewayDeployment } from "@cdktf/provider-aws/lib/api-gateway-deployment";
+import { ApiGatewayStage } from "@cdktf/provider-aws/lib/api-gateway-stage";
+import { ApiGatewayMethodSettings } from "@cdktf/provider-aws/lib/api-gateway-method-settings";
+import { LambdaPermission } from "@cdktf/provider-aws/lib/lambda-permission";
+import { DataAwsRegion } from "@cdktf/provider-aws/lib/data-aws-region";
+import { DataAwsCallerIdentity } from "@cdktf/provider-aws/lib/data-aws-caller-identity";
+
+import { Construct } from "constructs";
+import { core } from "..";
+import * as cloud from "../cloud";
+import { convertBetweenHandlers } from "../convert";
+import { Code } from "../core/inflight";
+import { Function } from "./function";
+import { ApiGatewayIntegration } from "@cdktf/provider-aws/lib/api-gateway-integration";
+import { IResource } from "../core";
+
+/**
+ * AWS Implementation of `cloud.Api`.
+ */
+export class Api extends cloud.ApiBase {
+  private readonly api: ApiGatewayRestApi;
+  constructor(scope: Construct, id: string, props: cloud.ApiProps = {}) {
+    super(scope, id, props);
+    this.api = new ApiGatewayRestApi(this, "api", {
+      name: id,
+    });
+  }
+
+  public get(
+    route: string,
+    inflight: core.Inflight,
+    props?: cloud.ApiOnGetProps
+  ): void {
+    const hash = inflight.node.addr.slice(-8);
+
+    const apiResource = new ApiGatewayResource(this, `resource-${hash}`, {
+      restApiId: this.api.id,
+      parentId: this.api.rootResourceId,
+      pathPart: route,
+    });
+
+    const apiMethod = new ApiGatewayMethod(this, `method-${hash}`, {
+      restApiId: this.api.id,
+      resourceId: apiResource.id,
+      httpMethod: "GET",
+      // FIXME: this cannot be hardcoded
+      authorization: "NONE",
+    });
+
+    const functionHandler = convertBetweenHandlers(
+      this.node.scope!, // ok since we're not a tree root
+      `${this.node.id}-OnGetRequestHandler-${hash}`,
+      inflight,
+      join(__dirname, "api.ongetrequest.inflight.js"),
+      "ApiOnGetRequestHandlerClient"
+    );
+    const fn = new cloud.Function(
+      this.node.scope!, // ok since we're not a tree root
+      `${this.node.id}-OnGetRequest-${hash}`,
+      functionHandler,
+      props
+    );
+    if (!(fn instanceof Function)) {
+      throw new Error("Queue only supports creating tfaws.Function right now");
+    }
+    const _tf_function: Function = fn;
+
+    new ApiGatewayIntegration(this, `integration-${hash}`, {
+      restApiId: this.api.id,
+      resourceId: apiResource.id,
+      httpMethod: apiMethod.httpMethod,
+      integrationHttpMethod: "POST",
+      type: "AWS_PROXY",
+      uri: _tf_function.invokeArn,
+    });
+
+    const region = new DataAwsRegion(this, "region");
```

Comment:
Let's refactor this a little bit so that the region information is provided by the `App` class. 

Here we want to have:
```suggestion
    const region = App.of(this).region;
```

To make that work, in `target-tf-aws/app.ts` we can add:

```ts
private awsRegionProvider: DataAwsRegion | undefined;
public get region(): string {
  if (!this.awsRegionProvider) {
    this.awsRegionProvider = new DataAwsRegion(this, "Region");
  }
  return this.awsRegionProvider.name;
}
```

You can also copy the implementation of `App.of` from `target-tf-azure/app.ts`